src/main/java/com/facebook/openwifirrm/modules/ProvMonitor.java

```diff
@@ -160,8 +194,79 @@ protected void syncDataToProv(
 			topo.values().stream().mapToInt(x -> x.size()).sum()
 		);
 
+		// Sync zone configs
+		deviceDataManager.updateZoneConfig(
+			configMap -> {
+				for (
+					Map.Entry<String, RRMDetails> entry : rrmDetails.entrySet()
+				) {
```

Comment:
You need to handle the case where some venue RRM config got newly wiped in Prov (and all venue devices were also deleted). So, before this loop, you should wipe zone config for all zones in `venueList.venues`.

This will also save you from needing to null out the venue config in the logic below if empty in Prov data, which you aren't doing now.